packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -269,9 +276,10 @@ export function createRootHooks<
     // request option should take priority over global
     const shouldAbortOnUnmount = opts?.trpc?.abortOnUnmount ?? abortOnUnmount;
 
-    const hook = __useQuery(
-      getArrayQueryKey(pathAndInput, 'query') as any,
-      (queryFunctionContext) => {
+    const hook = __useQuery({
+      ...ssrOpts,
+      queryKey: getArrayQueryKey(pathAndInput, 'query') as any,
+      queryFn: (queryFunctionContext) => {
```

Comment:
types complain here without the any - no idea why... either type the opts or the querykey as any 🤷🏼‍♂️

Reply:
meh 🤷  it's okay sometimes, feel free to leave a `// FIXME` comment if you want